pallets/pallet-bonded-coins/src/tests/transactions/start_destroy.rs

```diff
@@ -0,0 +1,412 @@
+use crate::{
+	mock::{runtime::*, *},
+	types::PoolStatus,
+	Error, Event, Pools,
+};
+use frame_support::{assert_err, assert_ok, traits::fungibles::Destroy};
+use frame_system::{pallet_prelude::OriginFor, RawOrigin};
+use sp_runtime::traits::BadOrigin;
+
+#[test]
+fn start_destroy_works() {
+	let pool_details = generate_pool_details(
+		vec![DEFAULT_BONDED_CURRENCY_ID],
+		get_linear_bonding_curve(),
+		true,
+		Some(PoolStatus::Active),
+		Some(ACCOUNT_00),
+		Some(DEFAULT_COLLATERAL_CURRENCY_ID),
+		Some(ACCOUNT_00),
+	);
+	let pool_id = calculate_pool_id(&[DEFAULT_BONDED_CURRENCY_ID]);
+
+	ExtBuilder::default()
+		.with_pools(vec![(pool_id.clone(), pool_details.clone())])
+		.with_native_balances(vec![(ACCOUNT_00, ONE_HUNDRED_KILT)])
+		.with_collaterals(vec![DEFAULT_COLLATERAL_CURRENCY_ID])
+		.build()
+		.execute_with(|| {
+			let origin = RawOrigin::Signed(ACCOUNT_00).into();
+
+			assert_ok!(BondingPallet::start_destroy(origin, pool_id.clone(), 1));
+
+			// Verify that the pool state has been updated to destroying
+			let updated_pool = Pools::<Test>::get(&pool_id).unwrap();
+			assert!(matches!(updated_pool.state, PoolStatus::Destroying));
```

Comment:
assert_eq! would look simpler. Same for the others further below

Reply:
1db6b8d7c70708533a100d0e28721449816fbe1a